admin/includes/classes/WhosOnline.php

```diff
@@ -133,12 +133,17 @@ public function retrieve($selectedView = '', $sessionToInspect = '', $exclude_sp
 
         $ip_array = [];
 
+        $sessionInspectAll = false; // Limit the session inspection to just the one $sessionToInspect
         // make the first entry show cart by default
         if (empty($sessionToInspect) && !$results->EOF) {
             $sessionToInspect = $results->fields['session_id']; // uses the fields array because we're not iterating the array yet
+            $sessionInspectAll = true; // No specific session was requested, so make all available.
         }
 
         foreach ($results as $result) {
+            if ($sessionInspectAll === true || !in_array($sessionToInspect, $results)) { // Use the session data from each session.
```

Comment:
I'm confused about the intention behind `|| !in_array($sessionToInspect, $results)`.

and can you further clarify the comment ` // Use the session data from each session.`?

Reply:
It was something I had tried to add in before realizing that there was an alternate way to address a non-existent session ***and*** that there was no need to individually track that a session had previously been "found" as they are expected to relatively be unique (or at least that was what I was thinking when I didn't do further development)...

On the comment, could you explain a little further about what clarification is desired? The idea behind that if statement is that if all sessions are to be "inspected" then within this loop either the $sessionToInspect value needs to be updated or
````php
            if (!empty($sessionToInspect) && $sessionToInspect === $result['session_id']) {
                $result['cart'] = $this->inspectSessionCart($result['session_id'], $result['session_data']);
            }
````
the code within the above test needs to be executed (which in a way is what the added code above "supports" because there is no code needed for additional empty session "testing".)

Perhaps focus more on collecting cart data? I've been working to fix the User Tracking plugin which review of this supported, so not entirely sure what the best verbiage is. I'm pretty sure I used the wrong words, but wanted to provide some semblance of explanation and as always welcome constructive feedback.

Reply:
If the `if` statement is simplified to just inspectAll === true, then there's probably no need for the comment.

I found the wording "from each session" confusing because we're already in a foreach, so why do we need to explain "each". 

Can the comment be skipped without losing necessary meaning? (espec. given the other two comments which already describe the new variable's purpose)

Reply:
Upon additional looking, fully agree... Comment can go away (choice of variable name offers some clarity) and yes that was the modification for the `if` statement.

Forced push coming up...